R/usage_to_annual_nets_distributed.R

```diff
@@ -0,0 +1,234 @@
+  
+  # Merge this with annual net distribution by country 
+  nets_distributed <- merge(x=nets_distributed, y=half_life_data[,c("iso3", "l")], 
+                            by = "iso3", all.x=TRUE, all.y=FALSE)
+  
+  # Use the net loss function to calculate th equilibrium nets per capita for the given
+  # annual nets distribution per capita by country
+  # This is for the specific "smoothed" case where we assume that 1/3 of the population at risk
+  # is distributed nets every 3 years (this is equivalent to a smoothed 3 yearly whole 
+  # population distribution).
+  nets_distributed$target_percapita_nets <- nets_distributed$annual_percapita_nets_distributed/ 
+    (integrate_net_loss_vector(k=k, l = nets_distributed$l) * (1/3))
+  
+  ### 2. Convert nets per capita to target usage ###
+  
+  # Use Loess curve to convert nets per capita to access
+  nets_distributed$target_access <- 
+    approx(x=loess_for_prediction$percapita_nets, 
+           y=loess_for_prediction$loess_predicted_access,
+           nets_distributed$target_percapita_nets)$y
+  
+  # Convert target access to target usage: usage = access * use rate
+  # since use rate = the proportion of people with access to a net who slept under it
+  # Relationship between access and NPC is aggregated for all countries,
+  # but use rate to convert usage to access is country-specific (derived from spatiotemporal regression?)
+  nets_distributed <- merge(x=nets_distributed, y=country_use_rate, 
+                            by = "iso3", all.x=TRUE, all.y=FALSE)
+  nets_distributed$target_usage <- nets_distributed$target_access*nets_distributed$use_rate
+  
+  nets_distributed <- nets_distributed[order(nets_distributed$iso3, nets_distributed$annual_percapita_nets_distributed),]
+  
+  return(nets_distributed[,c("iso3", "use_rate", "annual_percapita_nets_distributed",
+                             "target_percapita_nets","target_access", 
+                             "target_usage")])
+  
+  # TO DO:
+  # May need to extrapolate on the lower end of the curve - currently returning NAs
+  # What to do if percapita nets >1? Set to NA?
+  
+}
+
+#' Function of net loss over time
+#' Used to estimate the median net retention times by country.
+#' @param t Single value or vector of timesteps.
+#' @param k Fixed rate. Default = 20 from Bertozzi-Villa, Amelia, et al. Nature communications 12.1 (2021): 1-12.
+#' @param l Country-specific time at which no nets are retained (nets=0).
+#' @return The proportion of nets retained over time.
+#' @export
+net_loss <- function(t, k, l) {
+  prop_retained <- exp(k - k / (1 - (t / l) ^ 2))
+  prop_retained[t >= l] <- 0
+  return(prop_retained)
+}
+
+#' Integrate net loss function between 0 and 3
+#' @param k Fixed rate. Default = 20 from Bertozzi-Villa, Amelia, et al. Nature communications 12.1 (2021): 1-12.
+#' @param l Country-specific time at which no nets are retained (nets=0).
+#' @export
+integrate_net_loss <- function(k,l) {
```

Comment:
We probably want at least the upper to be an argument here  - we might want to assume other distribution intervals than 3 yearly.